sample/src/NimblePros.SampleToDo.Core/ProjectAggregate/Specifications/ProjectByIdWithItemsSpec.cs

```diff
@@ -2,7 +2,7 @@
 
 public class ProjectByIdWithItemsSpec : Specification<Project>
 {
-  public ProjectByIdWithItemsSpec(int projectId)
+  public ProjectByIdWithItemsSpec(ProjectId projectId)
```

Comment:
To avoid the `InvalidCastException`, use the strong type rather than the primitive type. When using the primitive type, EFCore tries to do a cast to the primitive type, which isn't needed if we just pass the strong type in the query.